src/Main.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from "react";
+import CreateGroup from './components/CreateGroup';
+import YourGroup from './components/YourGroup';
+import ConnectWallet from './components/ConnectWallet';
+
+import { ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+import { isWalletConnected, getGroupBalance, isMyTurn, isMemberOfGroup, isGroupOwner, whosTurnIsIt, 
+  getMyName, registerCallbackForEvents, getCurrentWalletAddress } from "./services/ethereumService";
+
+class Main extends Component {
+  state = { };
+
+  async componentDidMount() {
+    this.updateState();
+    registerCallbackForEvents(this.updateState);
+  }
+
+  updateState = async () => { 
+    this.setState({
```

Comment:
If we're going to poll every minute, I think it probably makes more sense to be more efficient with our data retrieval

i.e. maybe we set up a single public view on the smart contract that returns
- your group id (0 if you don't belong to one)
- Your entire group struct (which the front end ignores if the group id is 0)

And that way we can reduce the number of view calls per minute from 7 to 1

For demo purposes, this is totally fine, I'm just thinking long term.
I don't think we need to do that for this PR but I wanted to call it out anyways

Reply:
Agreed. I thought about this, but was more concerned about getting things done rather than optimizing.